File: frontend/src/store/api/result/dto/product/CommentStatistics.ts
export interface CommentStatistics {
    username: string,
    commentsCount: number,
    upvoteCount: number,
};

export function isCommentStatistics(value: any): value is CommentStatistics {
    return "commentsCount" in value && "upvoteCount" in value;
}

export function isCommentStatisticsArray(value: any[]): value is CommentStatistics[] {
    return value.every(isCommentStatistics);
}

File: backend/product-client/src/main/java/org/andante/product/dto/CommentStatistics.java
package org.andante.product.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
public class CommentStatistics {

    private String username;
    private Integer commentsCount;
    private Integer upvoteCount;
}
